src/eth/eip_155_transaction.rs

```diff
@@ -36,6 +42,104 @@ pub struct EthLegacyEip155TxArgs {
 }
 
 impl EthLegacyEip155TxArgs {
+    /// Returns legacy EIP-155 transaction signature
+    pub fn signature(&self, eth_chain_id: EthChainId) -> anyhow::Result<Signature> {
+        // Validate EIP155 Chain ID
+        validate_eip155_chain_id(eth_chain_id, &self.v)?;
+
+        // Convert r, s, v to byte arrays
+        let r_bytes = self.r.to_bytes_be().1;
+        let s_bytes = self.s.to_bytes_be().1;
+        let v_bytes = self.v.to_bytes_be().1;
+
+        // Pad r and s to 32 bytes
+        let mut sig = pad_leading_zeros(&r_bytes, 32);
+        sig.extend(pad_leading_zeros(&s_bytes, 32));
+
+        sig.extend(v_bytes);
+
+        // Prepend the one-byte legacy transaction marker
+        sig.insert(0, EIP_155_SIG_PREFIX);
```

Comment:
Wouldn't it make more sense to do it at the beginning to avoid re-shuffling?

Reply:
yes it will, fixed